README.md

```diff
@@ -227,6 +227,7 @@ manifest:
 
 > [!IMPORTANT]
 > Be sure to update your `config.yml` with the location of your `.synapseConfig` created in the step above, to avoid authentication errors. Paths can be specified relative to the `config.yml` file or as absolute paths.
+> If you are interacting with `schematic` with `python` directly and not through the CLI, the `.synapseConfig` needs to be in your current working directory to avoid authentication errors.
```

Comment:
This caveat is mentioned because of [this logic](https://github.com/Sage-Bionetworks/schematic/blob/90930cd876e88970ddea0ee1c0e28df371d823f2/schematic/store/synapse.py#L380) in the code.

To elaborate: the [`synapse_configuration_path`](https://github.com/Sage-Bionetworks/schematic/blob/1e39a05759fbeb7644d1e4582723beee49c16e33/schematic/configuration/configuration.py#L102) method always assumes [`_parent_directory`](https://github.com/Sage-Bionetworks/schematic/blob/1e39a05759fbeb7644d1e4582723beee49c16e33/schematic/configuration/configuration.py#L50) is the current working directory. This tells the python client to look for the `.synapseConfig` in whatever directory you're currently on, unless configured otherwise.

In this example I am repeating the logic [here](https://github.com/Sage-Bionetworks/schematic/blob/90930cd876e88970ddea0ee1c0e28df371d823f2/schematic/store/synapse.py#L380)
```
>>> import synapseclient
>>> syn = synapseclient.Synapse(configPath=os.getcwd(), cache_root_dir=os.getcwd())
>>> syn.login()

UPGRADE AVAILABLE

A more recent version of the Synapse Client (4.5.1) is available. Your version (4.3.1) can be upgraded by typing:
    pip install --upgrade synapseclient

Python Synapse Client version 4.5.1 release notes

https://python-docs.synapse.org/news/

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.venv/lib/python3.9/site-packages/synapseclient/client.py", line 753, in login
    raise SynapseNoCredentialsError("No credentials provided.")
synapseclient.core.exceptions.SynapseNoCredentialsError: No credentials provided.
```

And in this example, I remove the `configPath` and `cache_root_dir` which tells SynPy to assume `.synapseConfig` and `.synapseCache` are in the home directory, which is where they are created by default. This allowed me to log into Synapse despite not being in home:

```
>>> syn = synapseclient.Synapse()
>>> syn.login()

UPGRADE AVAILABLE

A more recent version of the Synapse Client (4.5.1) is available. Your version (4.3.1) can be upgraded by typing:
    pip install --upgrade synapseclient

Python Synapse Client version 4.5.1 release notes

https://python-docs.synapse.org/news/

Welcome, Jenny Medina!

INFO:synapseclient_default:Welcome, Jenny Medina!

>>> 
```